brave/brave-core
# feat(wallet): customize nonce in transaction panel

- show approve dialog in AssetDetails for unapproved tx
- Refactor service methods to use BaseActivity
- Fix Approve dialog hover issue


Resolves https://github.com/brave/brave-browser/issues/21459
Resolves 
Resolves 
Resolves 
Resolves 

## Submitter Checklist:


## Reviewer Checklist:


## After-merge Checklist:

  changes has landed on

## Test Plan:
1. Initiate a buy or swap transaction
2.  Go to advance setting
3. Find and enter a correct nonce value. usually it is the current largest transaction nonce value + 1 (required for valid Transaction submission) 
4. Click save and lock wallet
5. Unlock wallet and open transaction list, click on the previously created unapproved transaction (should show approved/reject dialog)
6. Validate the previously added nonce value
7. Approve the transactions and validate a successful submission of the transaction

For hover action buttons
1. On the transaction list screen (Portfolio => click on asset in list)
2. Initiate any swap transaction and go to the details section in approve/reject dialog then validate the alignments.

| Custom nonce  | Alignment |
| ----------- | ----------- |
|    ![custom-nonce](https://user-images.githubusercontent.com/11755381/157964655-8256543e-cd52-456f-9814-ae9d7d7501e8.mp4)   | ![approvereject](https://user-images.githubusercontent.com/11755381/157958360-f97fb249-433f-4bd5-b41b-7d4a2e344513.mp4) |
